dotnet/src/SemanticKernel.Abstractions/SkillDefinition/ISKFunction.cs

```diff
@@ -62,12 +63,14 @@ public interface ISKFunction
     /// </summary>
     /// <param name="input">String input</param>
     /// <param name="settings">LLM completion settings (for semantic functions only)</param>
+    /// <param name="memory">Semantic memory</param>
     /// <param name="logger">Application logger</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The updated context, potentially a new one if context switching is implemented.</returns>
     Task<SKContext> InvokeAsync(
         string? input = null,
         CompleteRequestSettings? settings = null,
+        ISemanticTextMemory? memory = null,
```

Comment:
in the current design, SKContext contains a Memory object. When calling this method, passing a single input string, one should be allowed to pass the current memory object stored in the kernel/context. This is not forcing to have a single memory, only allowing to carry over the existing memory. Without this, one could create a kernel with a memory instance, call a function passing a string, and the function internally would not find the memory defined earlier, e.g. "recall" would return no data. We had a few of these errors in the past in the notebooks.

Reply:
without this, SKFunction.InvokeAsync is creating a new context that doesn't have the memory configured by the user. For instance:

```
skill = kernel.importSkill(a skill that uses memory)
skill[functionname].invoke("this  content")
```

^^^ this will fail because the skill is designed to inject memory on call, not on ctor

And the change allows to do this:
```
memory1 = memory about me
memory2 = memory about someone

skill = kernel.importSkill(a skill that uses memory)
skill[functionname].invoke("this  content", memory1)
skill[functionname].invoke("this  content", memory2)
```

Reply:
gotcha. Makes sense, thank you